### DISCUSSION TITLE
A Genuine Question.

### DISCUSSION THREAD

**Discussion by DinoAwakening** [keiwando/toonsquid#605] [General]
For any programmer who might read this in the forum: How long would it actually take to port ToonSquid to Android. I mean, it's an app like no other. And I imagine some of us are struggling finding a good animation or even a drawing app equivalente to ToonSquid. Nevertheless, just as mentioned before, it's an app like no other, which implies there's not an app available for Android that works the same way ToonSquid does... :/

**Comment by Lioncub901** [comment]
Great Question. 
Hi, my name is Israel and I program games on my iPad using Codea which uses Objective C and Apple libraries as part of the app. 

Short Answer: many years (3 - 6 years), it is a difficult task which is not worth it for most devs especially one man devs.

Long Answer:
I will say that it is no easy task to port a IOS app to any platform. Apple tends to be very restrictive and does not really allow their apps to be ported to other platform. If a person wants to port an IOS app/game to other platforms the person has to rewrite the whole code to the other platform which majority of dev especially a one man devs does not want to do. Also you have to keep in mind what profit will be made if one ports to a different device. Is it worth it? IOS app tend to make 5x times the amount of money compared to Android counterparts. Most android users are not willing to buy paid apps but would rather watch ads which some dev don’t like since they do reduce the quality of the app and kind of distracts away from the app. So if the ToonSquid dev happens to make a port to Android with a price of $10 most android users will not pay for it but rather pirate which explains why IOS apps/games make 5 times more money. Apple makes pirating near impossible and punish users that do, which is not great for users, but it give more security for the developers which is why majority of devs prefer to go to IOS. Making apps/games for IOS is safer and makes more money (the devs do need to eat). You also have to keep account that there are about 8 IOS devices devs have to support, all usually running the same operating system version, while for Android there are over 1000 devices all with different sizes and different operating system versions. If the ToonSquid dev ports to Android he would need to support all of those devices and operating system version which is a lot work for one person. Making one app which just works on a few devices running simliar versions of the operating system is much easier. So to most devs it is not really worth it to port to Android. But if ToonSquid get more popular and the dev hires other people then he might port to Android but the Android version would probably be a second thought.

**Reply by DinoAwakening** [reply]
OMG! Thanks for your answer and for explaining it in such details! That's quite sad that I have to agree with you in terms of compatibility and worth of an android app. There's plenty people with unknown android models which might not have the usual aspecto ratio or even same android version or ROM at all. Also most people, like you said, would rather to crack any app that isn't free, and spread the .APK among their closest ones or YouTube perhaps. Which end up not being worth it... :/

Again, thanks for your response! Willing to know what's next with ToonSquid :D

**Reply by Lioncub901** [reply]
You’re very welcome, it is a shame since it does seem like you want to pay. We just don’t have the technology yet to convert IOS apps to Android, actually I would rather say that greed is preventing the advancement of that technology from both the customers and the companies side (everyone wants to do what benefits themselves). But don’t give up hope there is Krita on the Android (I wish it was on IOS) it has a great animation system and it FREEEEEE!!! There is also rough animator might not have the best ui but it powerful and is a cheap one time purchase. There is also Clip Studio Paint (I use this a lot) though it is a subscription 😔 it is in my opinion the best art/animation app ever. You might want to consider it when you get to an intermediate/advance level at digital art. But remember it is not the tools that make an artwork great but the skill of the artist. A guy made a realistic looking [Santa](https://www.youtube.com/watch?v=v2g5qbvb7F4) in Microsoft Paint so you can make any animation you want with the right skill so keep on Animating.
